# Assignment 5

Assume you work for a record company. Your boss is interested in predicting ablum sales based on advertising. You are provided with the data in a file (*album_sales.csv*). 

There are 200 rows, where each row represents an album. There are also four columns. The first (*sales*) represents the sales of each album (in thousands) the week after release. The second (adverts) represents the amount (in thousands $) spent promoting the album prior to its release. The third (*airplay*) represents how many times it played on the radio. The fourth column (*attract*) represents the attractiveness of the band. 

## Instructions

### 1. Visualization (15 points)

Visualizing data is important to get a 'feel' of what the relationship between variables could be. For this part, **display the scatterplots between the following variables**.

- *sales* and *adverts*
- *sales* and *airplay*
- *sales* and *attract*

In [ ]:
import statsmodels.api as sm
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

df = pd.read_csv("album_sales.csv")
df.head()

In [ ]:
# 1.) sales and adverts
x = df['sales']
y = df.adverts
x = sm.add_constant(x)

model = sm.OLS(y, x).fit()
predictions = model.predict(x)
print(model.summary())

plt.scatter(x['sales'], y, alpha=0.3)
plt.xlabel('sales')
plt.ylabel('adverts')
plt.plot(x['sales'], predictions, color='red')
plt.show()

In [ ]:
# 2.) sales and airplay
x = df['sales']
y = df.airplay
x = sm.add_constant(x)

model = sm.OLS(y, x).fit()
predictions = model.predict(x)
print(model.summary())

plt.scatter(x['sales'], y, alpha=0.3)
plt.xlabel('sales')
plt.ylabel('airplay')
plt.plot(x['sales'], predictions, color='green')
plt.show()

In [ ]:
# 3.) sales and attract
x = df['sales']
y = df.attract
x = sm.add_constant(x)

model = sm.OLS(y, x).fit()
predictions = model.predict(x)
print(model.summary())

plt.scatter(x['sales'], y, alpha=0.3)
plt.xlabel('sales')
plt.ylabel('attract')
plt.plot(x['sales'], predictions, color='blue')
plt.show()

In the empty `Markdown` cell below, describe the relationship (or lack thereof) portrayed by the scatterplots you generated. 

1. The dots show a positive relationship because as sales increases up, adverts increases and the line is going upwards, showing a positive correlation. This scatterplot has the highest coreelation. 
 
2. The dots show a positive relationship because as sales increases, airplay increases, and the line is going upwards, showing a positive correlation. This scatterplot has a high coreelation. 
 
3. The dots show a weak positive relationship because as sales increases, attract increases (even if it's little), and the line is going upwards, but by a small angle. Compared to the other two, this scatterplot has the weakest coreelation. 

### 2. Linear Regression (35 points)

- Conduct a linear regression to construct a linear model between *sales* and *adverts*
- Display the **F-statistic** and **P-value**
    - Use the `statsmodels.api` module for this -- may need to install via `pip`
    - Note that **X**, your predictor(s), will consist of *adverts* and that **y**, your constant, will consist of *sales*
    - The values you need can be accessed via the `summary()` function of your model object

In [2]:
# 1.) sales and adverts
x = df['sales']
y = df.adverts
x = sm.add_constant(x)

model = sm.OLS(y, x).fit()
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                adverts   R-squared:                       0.335
Model:                            OLS   Adj. R-squared:                  0.331
Method:                 Least Squares   F-statistic:                     99.59
Date:                Mon, 21 Sep 2026   Prob (F-statistic):           2.94e-19
Time:                        11:12:16   Log-Likelihood:                -1479.6
No. Observations:                 200   AIC:                             2963.
Df Residuals:                     198   BIC:                             2970.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        -58.1948     73.016     -0.797      0.4

In the empty `Markdown` cell below, describe whether the **F-statistic** and **P-value** describe about our linear regression model. (I.e., Is it good? Bad? No bearing?)

The F-statistic is 99.59 and the p-value is 2.94e-19, these mean that the linear regression model is good because the sales is a meaningful predictor of adverts.

### 3. Model Coefficients (10 points)

- Make note of the **intercept** value and **coefficient** (adverts) value of your linear regression model. 

The regression line is described using this equation: *Y = b<sub>0</sub> + b<sub>1</sub>X<sub>1</sub>* where each variable represents the following:

- Y denotes the sales
- *b<sub>0</sub>* denotes the **intercept** value
- *b<sub>1</sub>* denotes the advertising budget **coefficient**
- *X<sub>2</sub>* denotes the advertising budget

The equation then becomes *Album Sales = intercept value + (coefficient * advertising budget)*. 

Using the intercept value and coefficient of your linear model, **calculate and display how many records will be sold if $135,000 was spent on advertising an album**.

In [6]:
x = df['adverts']
y = df['sales']

# Add constant for intercept
x = sm.add_constant(x)

model = sm.OLS(y, x).fit()

prediction = model.predict([1, 135])
print(prediction)


[147.11674342]


### 4. Multiple Regression (40 points)
- Conduct a **multiple regression** to construct a model between *sales* and the predictors (*adverts*, *airplay*, and *attract*)
    - Note that **X**, your predictors will consist of *adverts*, *airplay*, and *attractiveness*, and **y**, your constant, will consist of *sales*
- Display the summary of the multiple regression model using the `summary()` function to obtain the relevant values

In [ ]:
x = df[['adverts', 'airplay', 'attract']]
y = df.sales
x = sm.add_constant(x)

model = sm.OLS(y, x).fit()
predictions = model.predict(x)
print(model.summary())

                            OLS Regression Results                            
Dep. Variable:                  sales   R-squared:                       0.665
Model:                            OLS   Adj. R-squared:                  0.660
Method:                 Least Squares   F-statistic:                     129.5
Date:                Mon, 21 Sep 2026   Prob (F-statistic):           2.88e-46
Time:                        11:20:35   Log-Likelihood:                -1052.2
No. Observations:                 200   AIC:                             2112.
Df Residuals:                     196   BIC:                             2126.
Df Model:                           3                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const        -26.6130     17.350     -1.534      0.1

We know that the R<sup>2</sup> (R-Squared) value can be used to evaluate the overall fit of a linear model. We also know that R<sup>2</sup> is between 0 and 1 -- higher values are better because it means that more variance is explained by the model. **Higher R<sup>2</sup> values are better if their P-values are < 0.05**.

Bsed on this, discuss **which one of the two models you constructed is better** in the empty `Markdown` cell below. 
* Model 1: The linear model between *sales* and *adverts* you constructed
* Model 2: The multiple regression model between (*sales*) and the predictors (*adverts*, *airplay*, and *attract*) you constructed

The best model is model 2, or the multiple regression model, because R^2 is higher than the first model and all the predictors have significant p-values.

### Submission
- Submit this `Jupyter` file to D2L, renamed as **Last_First_Assignment4.ipynb** 
    - Replace '**Last**' and '**First**' with your first and last name